src/nurgling/actions/HarvestCrop.java

```diff
@@ -166,9 +179,16 @@ void harvest(NGameUI gui, HashMap<Gob,AtomicBoolean> barrelInfo, Gob trough, Gob
                 new TransferToTrough(trough, iseed, cistern).run(gui);
             }
         }
-        if(NUtils.getStamina()<0.35)
+        if(NUtils.getStamina()<0.35) {
+            if (prop.autorefill) {
+                if (FillWaterskins.checkIfNeed())
```

Comment:
This isnt ideal since it checks if a refill is required on every drink. We can just attempt a refill on a failed drink instead below and attampt to drink again.